app/models/Extraction.scala

```diff
@@ -122,7 +123,7 @@ object ExtractorInfo {
   }
   implicit val dateFormat = new Format[Date] {
     def reads(json: JsValue): JsResult[Date] = JsSuccess(json.as[Date])
-    def writes(date: Date): JsValue = Json.toJson(date.toString)
+    def writes(date: Date): JsValue = Json.toJson(date)
```

Comment:
I guess we don't usually perform updates via the API, but this was throwing a parse exception for me. Upon closer inspection, it looked like the `updated` date string was double-quoted.

Reply:
Were you including the `updated` field when posting? I believe that's managed internally and should not be included in the post (we should validate for it). 

With this change, I am seeing the post still failing if I include the `updated` field but when reading the results back out with `GET api/extractors` I get the epoch date now, for example `"updated": 1601317874506,`.

Reply:
I wasn't planning on removing any fields before sending it back to the backend. I am simply sending back the same object that we receive from the API with a slight modification. I would argue that if this field is important to Clowder internals, then the backend should ignore it if the user sends it back during an update. Would we allow the user to change the `_id` in the same manner?

Removing the `.toString` was meant to just remove the repeated double-quotes, but I guess it transformed this value into the epoch time as well? Apparently I'm not very good in dealing with Scala's `reads`/`writes` encoder logic. I would think that epoch time would be preferable in this case, to avoid needing to parse dates differently between Scala / Javascript, but maybe we prefer the string timestamps instead?

Reply:
We do want to send the update date to the client. I am not sure we want to let the client specify one. Since it is when this document was last updated, it makes sense for the server to set that.

Either way, I wasn't able to understand what this was fixing. This is what I saw in behavior when comparing change with no change: "With this change, I am seeing the post still failing if I include the updated field but when reading the results back out with GET api/extractors I get the epoch date now, for example "updated": 1601317874506,." Am I doing something wrong? I don't think we want to send `updated": 1601317874506,.` to the client (vs what it currently does which is format the date).

Reply:
I added this line because without it, I was getting parse problems sending ExtractorInfo back to the API.

Test it out yourself on Clowder:
```
# GET an ExtractorInfo model from the API
MacBook-Pro:clowder lambert8$ curl --cookie 'id=<REDACTED>' https://dev.simpl.syngentadigitalapps.com/clowder/api/extractors/ncsa.image.preview > extractor.json
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   657  100   657    0     0   2783      0 --:--:-- --:--:-- --:--:--  2783

# POST the exact same object back to the API (this is how the UI performs updates)
MacBook-Pro:clowder lambert8$ curl -XPOST --cookie 'id=<REDACTED>' https://dev.simpl.syngentadigitalapps.com/clowder/api/extractors -H "Content-Type: application/json" -d @./extractor.json
{"status":"error","request":"POST /clowder/api/extractors","exception":"@7ha02hmdc: Execution exception in null:null\\n\tat play.api.Application$class.handleError(Application.scala:293)\\n\tat play.api.DefaultApplication.handleError(Application.scala:399)\\n\tat play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$3.apply(PlayDefaultUpstreamHandler.scala:268)\\n\tat play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$3.apply(PlayDefaultUpstreamHandler.scala:268)\\n\tat scala.Option.map(Option.scala:145)\\n\tat play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:268)\\n\tat play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:264)\\n\tat scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:344)\\n\tat scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:343)\\n\tat scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)\\n\tat play.api.libs.iteratee.Execution$$anon$1.execute(Execution.scala:43)\\n\tat scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:40)\\n\tat scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:248)\\n\tat scala.concurrent.Promise$class.complete(Promise.scala:55)\\n\tat scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:153)\\n\tat scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:249)\\n\tat scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:249)\\n\tat scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)\\n\tat play.api.libs.iteratee.Execution$$anon$2.execute(Execution.scala:70)\\n\tat scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:40)\\n\tat scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:248)\\n\tat scala.concurrent.Promise$class.complete(Promise.scala:55)\\n\tat scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:153)\\n\tat scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:249)\\n\tat scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:249)\\n\tat scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)\\n\tat play.api.libs.iteratee.Execution$$anon$2.execute(Execution.scala:70)\\n\tat scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:40)\\n\tat scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:248)\\n\tat scala.concurrent.Promise$class.complete(Promise.scala:55)\\n\tat scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:153)\\n\tat scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:249)\\n\tat scala.concurrent.Future$$anonfun$flatMap$1.apply(Future.scala:249)\\n\tat scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)\\n\tat play.api.libs.iteratee.Execution$$anon$2.execute(Execution.scala:70)\\n\tat scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:40)\\n\tat scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:248)\\n\tat scala.concurrent.Promise$class.complete(Promise.scala:55)\\n\tat scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:153)\\n\tat scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:23)\\n\tat akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:42)\\n\tat akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:386)\\n\tat scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)\\n\tat scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)\\n\tat scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)\\n\tat scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)\\nCaused by: play.api.libs.json.JsResultException: JsResultException(errors:List((,List(ValidationError(error.expected.date.isoformat,WrappedArray(yyyy-MM-dd))))))\\n\tat play.api.libs.json.JsValue$$anonfun$2.apply(JsValue.scala:64)\\n\tat play.api.libs.json.JsValue$$anonfun$2.apply(JsValue.scala:64)\\n\tat play.api.libs.json.JsResult$class.fold(JsResult.scala:71)\\n\tat play.api.libs.json.JsError.fold(JsResult.scala:10)\\n\tat play.api.libs.json.JsValue$class.as(JsValue.scala:62)\\n\tat play.api.libs.json.JsString.as(JsValue.scala:117)\\n\tat models.ExtractorInfo$$anon$2.reads(Extraction.scala:124)\\n\tat play.api.libs.json.PathReads$$anonfun$at$1$$anonfun$apply$2.apply(JsConstraints.scala:33)\\n\tat play.api.libs.json.PathReads$$anonfun$at$1$$anonfun$apply$2.apply(JsConstraints.scala:33)\\n\tat play.api.libs.json.JsResult$class.flatMap(JsResult.scala:97)\\n\tat play.api.libs.json.JsSuccess.flatMap(JsResult.scala:6)\\n\tat play.api.libs.json.PathReads$$anonfun$at$1.apply(JsConstraints.scala:33)\\n\tat play.api.libs.json.PathReads$$anonfun$at$1.apply(JsConstraints.scala:33)\\n\tat play.api.libs.json.Reads$$anon$8.reads(Reads.scala:98)\\n\tat play.api.libs.json.Reads$$anonfun$orElse$1.apply(Reads.scala:47)\\n\tat play.api.libs.json.Reads$$anonfun$orElse$1.apply(Reads.scala:47)\\n\tat play.api.libs.json.Reads$$anon$8.reads(Reads.scala:98)\\n\tat play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:78)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anon$8.reads(Reads.scala:98)\\n\tat play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:78)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anon$8.reads(Reads.scala:98)\\n\tat play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:78)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anon$8.reads(Reads.scala:98)\\n\tat play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:78)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anon$8.reads(Reads.scala:98)\\n\tat play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:78)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anon$8.reads(Reads.scala:98)\\n\tat play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:78)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anon$8.reads(Reads.scala:98)\\n\tat play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:78)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anon$8.reads(Reads.scala:98)\\n\tat play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:78)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anon$8.reads(Reads.scala:98)\\n\tat play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:78)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anon$8.reads(Reads.scala:98)\\n\tat play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:78)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anon$8.reads(Reads.scala:98)\\n\tat play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:78)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anon$8.reads(Reads.scala:98)\\n\tat play.api.libs.json.Reads$$anon$3$$anon$4.reads(Reads.scala:78)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anonfun$map$1.apply(Reads.scala:25)\\n\tat play.api.libs.json.Reads$$anon$8.reads(Reads.scala:98)\\n\tat play.api.libs.json.JsValue$class.validate(JsValue.scala:70)\\n\tat play.api.libs.json.JsObject.validate(JsValue.scala:163)\\n\tat api.Extractions$$anonfun$addExtractorInfo$1.apply(Extractions.scala:469)\\n\tat api.Extractions$$anonfun$addExtractorInfo$1.apply(Extractions.scala:459)\\n\tat play.api.mvc.ActionBuilder$$anonfun$apply$10.apply(Action.scala:221)\\n\tat play.api.mvc.ActionBuilder$$anonfun$apply$10.apply(Action.scala:220)\\n\tat api.ApiController$$anon$4.invokeBlock(ApiController.scala:65)\\n\tat play.api.mvc.ActionBuilder$$anon$1.apply(Action.scala:309)\\n\tat play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4$$anonfun$apply$5.apply(Action.scala:109)\\n\tat play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4$$anonfun$apply$5.apply(Action.scala:109)\\n\tat play.utils.Threads$.withContextClassLoader(Threads.scala:18)\\n\tat play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4.apply(Action.scala:108)\\n\tat play.api.mvc.Action$$anonfun$apply$1$$anonfun$apply$4.apply(Action.scala:107)\\n\tat scala.Option.map(Option.scala:145)\\n\tat play.api.mvc.Action$$anonfun$apply$1.apply(Action.scala:107)\\n\tat play.api.mvc.Action$$anonfun$apply$1.apply(Action.scala:100)\\n\tat play.api.libs.iteratee.Iteratee$$anonfun$mapM$1.apply(Iteratee.scala:481)\\n\tat play.api.libs.iteratee.Iteratee$$anonfun$mapM$1.apply(Iteratee.scala:481)\\n\tat play.api.libs.iteratee.Iteratee$$anonfun$flatMapM$1.apply(Iteratee.scala:517)\\n\tat play.api.libs.iteratee.Iteratee$$anonfun$flatMapM$1.apply(Iteratee.scala:517)\\n\tat play.api.libs.iteratee.Iteratee$$anonfun$flatMap$1$$anonfun$apply$14.apply(Iteratee.scala:494)\\n\tat play.api.libs.iteratee.Iteratee$$anonfun$flatMap$1$$anonfun$apply$14.apply(Iteratee.scala:494)\\n\tat scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)\\n\tat scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)\\n\t... 6 more\\n"}
```

> We do want to send the update date to the client. I am not sure we want to let the client specify one. Since it is when this document was last updated, it makes sense for the server to set that. 

Again, I agree that the client should not be allowed to set this value. The short of it is that the backend has **never** received an `updated` field on an ExtractorInfo from the client before now, as this field is not included in `extractor_info.json`. Now that we are sending this field back, the backend **needs** to ignore it when performing the update. Assuming that users will strip out fields before sending back resources is not going to be a very safe assumption.

> I don't think we want to send updated": 1601317874506,. to the client (vs what it currently does which is format the date). 

I honestly don't think the client cares about the format at this point, but I disagree. We absolutely **should** be sending epoch time imo because that is a format that can be read by both Scala and Javascript. We are formatting the date for America/Chicago and then sending it as a String, when the client's browser could easily localize that for us in their own timezone. I'm confused what the benefit of having it as a String is when 99% of clients and users aren't even going to look at this field, and many of the ones who do could see the wrong timezone.

Reply:
Thank you, I understand now and was able to reproduce. We can merge this as is. I have a fix coming that ignores the date when posting and formats the date formatted with ISO08601 coming. Made an issue for it https://github.com/clowder-framework/clowder/issues/89.